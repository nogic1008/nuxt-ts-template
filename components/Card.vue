<template>
  <div class="column">
    <div class="card">
      <header class="card-header">
        <p class="card-header-title has-text-grey">
          {{ title }}
        </p>
      </header>
      <div class="card-content">
        <div class="content has-text-centered">
          <b-icon :icon="icon" size="is-large" type="is-primary" />
        </div>
      </div>
      <footer class="card-footer">
        <div class="card-footer-item">
          <span>
            <slot />
          </span>
        </div>
      </footer>
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'nuxt-property-decorator'

@Component
export default class CardComponent extends Vue {
  @Prop({ required: true, type: String })
  readonly title!: string

  @Prop({ required: true, type: String })
  readonly icon!: string
}
</script>
