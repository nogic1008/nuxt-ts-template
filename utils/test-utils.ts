/** Generate [0-9a-z]{length} string. */
export const generateRandomString = (length: number) =>
  [...Array(Number.isInteger(length) ? length : 12)]
    .map(() => (~~(Math.random() * 36)).toString(36))
    .join('')
