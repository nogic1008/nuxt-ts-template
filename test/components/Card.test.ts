import { createLocalVue, mount, shallowMount, Wrapper } from '@vue/test-utils'
import Buefy from 'buefy'

import Card from '~/components/Card.vue'

const localVue = createLocalVue()
localVue.use(Buefy)

describe('components/Card.vue', () => {
  let wrapper: Wrapper<Vue>
  const props = {
    title: 'title',
    icon: 'github-circle'
  }

  beforeEach(() => {
    wrapper = shallowMount(Card, { localVue, propsData: { ...props } })
  })

  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders correctly', () => {
    const wrapper = mount(Card, { localVue, propsData: { ...props } })
    expect(wrapper.element).toMatchSnapshot()
  })

  test('renders props value', () => {
    expect(wrapper.find('.card-header-title').text()).toBe(props.title)
    expect(wrapper.find('b-icon-stub').attributes('icon')).toBe(props.icon)
  })
})
