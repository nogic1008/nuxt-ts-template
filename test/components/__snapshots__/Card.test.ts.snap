// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Card.vue renders correctly 1`] = `
<div
  class="column"
>
  <div
    class="card"
  >
    <header
      class="card-header"
    >
      <p
        class="card-header-title has-text-grey"
      >
        
        title
      
      </p>
    </header>
     
    <div
      class="card-content"
    >
      <div
        class="content has-text-centered"
      >
        <span
          class="icon has-text-primary is-large"
        >
          <i
            class="mdi mdi-github-circle mdi-48px"
          />
        </span>
      </div>
    </div>
     
    <footer
      class="card-footer"
    >
      <div
        class="card-footer-item"
      >
        <span />
      </div>
    </footer>
  </div>
</div>
`;
